mcs/class/corlib/System.Reflection/AssemblyName.cs

```diff
@@ -84,17 +85,30 @@ public AssemblyName ()
 		}
 
 		[MethodImpl (MethodImplOptions.InternalCall)]
-		static extern bool ParseName (AssemblyName aname, string assemblyName);
-		
+		static extern bool ParseAssemblyName (IntPtr name, out MonoAssemblyName aname, out bool is_version_definited, out bool is_token_defined);
+
 		public AssemblyName (string assemblyName)
 		{
 			if (assemblyName == null)
 				throw new ArgumentNullException ("assemblyName");
 			if (assemblyName.Length < 1)
 				throw new ArgumentException ("assemblyName cannot have zero length.");
-				
-			if (!ParseName (this, assemblyName))
-				throw new FileLoadException ("The assembly name is invalid.");
+
+
+			MonoAssemblyName nativeName = default (MonoAssemblyName);
```

Comment:
You don't need default it should be fully assigned and I'd move it just before ParseAssemblyName